kiwi/container/docker.py

```diff
@@ -39,14 +62,65 @@ def create(self, filename):
         :param string filename: archive file name
         """
         exclude_list = [
-            'image', '.profile', '.kconfig', 'boot',
+            'image', '.profile', '.kconfig', 'boot', 'dev', 'sys', 'proc',
             Defaults.get_shared_cache_location()
         ]
-        # replace potential suffix from filename because
-        # it is added by the archive creation call
-        archive = ArchiveTar(
-            filename.replace('.xz', '')
+
+        self.docker_dir = mkdtemp(prefix='kiwi_docker_dir.')
+        self.docker_root_dir = mkdtemp(prefix='kiwi_docker_root_dir.')
+
+        container_dir = os.sep.join(
+            [self.docker_dir, 'umoci_layout']
+        )
+        container_tag = ':'.join(
+            [container_dir, self.container_name]
+        )
+
+        Command.run(
+            ['umoci', 'init', '--layout', container_dir]
         )
-        archive.create_xz_compressed(
-            source_dir=self.root_dir, exclude=exclude_list
+        Command.run(
+            ['umoci', 'new', '--image', container_tag]
         )
+        Command.run(
+            ['umoci', 'unpack', '--image', container_tag, self.docker_root_dir]
+        )
+        docker_root = DataSync(
+            self.root_dir + os.sep, os.sep.join([self.docker_root_dir, 'rootfs'])
+        )
+        docker_root.sync_data(
+            options=['-a', '-H', '-X', '-A'], exclude=exclude_list
+        )
+        Command.run(
+            ['umoci', 'repack', '--image', container_tag, self.docker_root_dir]
+        )
+        Command.run(
+            [
+                'umoci', 'config',
+                '='.join(['--config.cmd', self.entry_command]),
```

Comment:
Yes this is quite strange to my eyes also. I assume it is the same as using the CMD statement of the Dockerfile using the brackets syntax (i.e CMD ['/usr/bin/cowsay', 'have', 'fun']), in that case every parameter is passed to the `exec` function instead to a shell.